static/js/compose.js

```diff
@@ -724,13 +724,49 @@ function validate_stream_message() {
     return true;
 }
 
+//The function will check whether the recipient is in settings.CROSS_REALM_BOT_EMAILS or settings.ADMIN_DOMAIN
 function validate_private_message() {
     if (exports.recipient() === "") {
         compose_error("Please specify at least one recipient", $("#private_message_recipient"));
         return false;
     }
+    else {
+        var private_recipients = exports.recipient().split(/\s*[,;]\s*/);
+        var invalid_recipients = "";
+        var invalid_recipients_count = 0;
+        _.each(private_recipients, function (email) {
+            //This case occurs when exports.recipient() ends with ','
+            if(email === "") {
+                return;
+            }
+            if(page_params.cross_realm_bot_emails.indexOf(email) !== -1) {
```

Comment:
@timabbott Can you explain a little bit more on using dictionary and fold_case to do the case-insensitive comparison? Also what should be the name of the function when we refactor the code.